notes/stv/20230125-red-deploy-01.txt

```diff
@@ -0,0 +1,952 @@
+# Copy certificates from data server.
+#[root@ansibler]        
+
+scp -r fedora@data.gaia-dmp.uk:/home/fedora/certs/ /root/
+certs.tar.gz                                                                                                                                                               0%    0     0.0KB/s   --:-- ETA
+certs.tar.gz                                                                                                                                                             100%   22KB  92.0KB/s   00:00    
+dmp.gaia.ac.uk.tar.gz                                                                                                                                                    100% 9822    55.8KB/s   00:00    
+zeppelin.gaia-dmp.uk.tar.gz                                                                                                                                              100%   50KB 196.7KB/s   00:00    
+iris-gaia-red.gaia-dmp.uk.tar.gz                                                                                                                                         100% 9635    63.6KB/s   00:00    
+
+
+
+
+# -----------------------------------------------------
+# Get the public IP address of our Zeppelin node.
+#[root@ansibler]
+[
+
+
+    deployname=$(
+        yq eval \
+            '.aglais.status.deployment.name' \
+            '/opt/aglais/aglais-status.yml'
+        )
+
+    zeppelinid=$(
+        openstack \
+            --os-cloud "${cloudname:?}" \
+            server list \
+                --format json \
+        | jq -r '.[] | select(.Name == "'${deployname:?}'-zeppelin") | .ID'
+        )
+
+    zeppelinip=$(
+        openstack \
+            --os-cloud "${cloudname:?}" \
+            server show \
+                --format json \
+                "${zeppelinid:?}" \
+        | jq -r ".addresses | .\"${deployname}-internal-network\" | .[1]"
+        )
+
+cat << EOF
+Zeppelin ID [${zeppelinid:?}]
+Zeppelin IP [${zeppelinip:?}]
+EOF
+
+Zeppelin ID [deab239f-8285-411c-97ae-bd5cb08eae75]
+Zeppelin IP [128.232.222.163]
+    
+    
+    
+    
+
+# --------------------------------------------
+# Update the dns entry with new IP
+# dmp.gaia.ac.uk -> 128.232.222.163
+
+    # This should be done automatically.
+    # https://github.com/wfau/aglais/issues/893
+    source /deployments/zeppelin/bin/create-user-tools.sh
+
+    ducktoken=$(getsecret 'devops.duckdns.token')
+
+    duckname=aglais-live
+
+    curl "https://www.duckdns.org/update/${duckname:?}/${ducktoken:?}/${zeppelinip:?}"    
+
+    > OK
+    
+    
+    
+# -----------------------------------------------------
+# Enable HTTPS
+#[root@ansibler]
+
+    /deployments/hadoop-yarn/bin/setup-ssl.sh \
+        "${cloudname:?}" \
+        "${configname:?}" \
+    | tee /tmp/setup-ssl.log
+
+
+# Failed
+
+# Looks like the hostname "zeppelin.gaia-dmp.uk" ended up in the nginx config, where the SSL certificate is under "dmp.gaia.ac.uk"
```

Comment:
If we find something that doesn't work, we shouldn't just leave it. This should either be fixed, or at a minimum logged as an issue to fix later.

Reply:
Issue now created here: https://github.com/wfau/gaia-dmp/issues/1106